File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    stream_from "ChatChannel"
  end

  def receive(msg)
    logger.info(msg)
    ActionCable.server.broadcast('ChatChannel', msg)
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end

File: app/javascript/channels/chat_channel.js
import { createCable } from "@anycable/web";

const cable = createCable();

async function run() {
  const channel = await cable.subscribeTo("ChatChannel")
  const _ = await channel.perform("receive")
  channel.on("message", (msg) => {
    console.log(msg)
  })
}

run()